tests/integ/test_mxnet_train.py

```diff
@@ -72,6 +72,47 @@ def test_deploy_model(mxnet_training_job, sagemaker_session):
         predictor.predict(data)
 
 
+def test_deploy_model_with_update_endpoint(mxnet_training_job, sagemaker_session):
+    endpoint_name = 'test-mxnet-deploy-model-{}'.format(sagemaker_timestamp())
+
+    with timeout_and_delete_endpoint_by_name(endpoint_name, sagemaker_session):
+        desc = sagemaker_session.sagemaker_client.describe_training_job(TrainingJobName=mxnet_training_job)
+        model_data = desc['ModelArtifacts']['S3ModelArtifacts']
+        script_path = os.path.join(DATA_DIR, 'mxnet_mnist', 'mnist.py')
+        model = MXNetModel(model_data, 'SageMakerRole', entry_point=script_path,
+                           py_version=PYTHON_VERSION, sagemaker_session=sagemaker_session)
+        model.deploy(1, 'ml.t2.medium', endpoint_name=endpoint_name)
+        old_production_variants = sagemaker_session.describe_endpoint(EndpointName=endpoint_name)['ProductionVariants']
+
+        model.deploy(1, 'ml.m4.xlarge', update_endpoint=True, endpoint_name=endpoint_name)
+        new_production_variants = sagemaker_session.describe_endpoint(EndpointName=endpoint_name)['ProductionVariants']
+
+        assert new_production_variants['InstanceType'] == 'ml.m4.xlarge'
+        assert old_production_variants['VariantName'] != new_production_variants['VariantName']
+        assert old_production_variants['ModelName'] == new_production_variants['ModelName']
+        assert old_production_variants['InitialInstanceCount'] == new_production_variants['InitialInstanceCount']
+        assert old_production_variants['InitialVariantWeight'] == new_production_variants['InitialVariantWeight']
+        assert old_production_variants['AcceleratorType'] == new_production_variants['AcceleratorType']
+
+
+def test_deploy_model_with_update_non_existing_endpoint(mxnet_training_job, sagemaker_session):
+    endpoint_name = 'test-mxnet-deploy-model-{}'.format(sagemaker_timestamp())
+
+    with timeout_and_delete_endpoint_by_name(endpoint_name, sagemaker_session):
+        try:
+            desc = sagemaker_session.sagemaker_client.describe_training_job(TrainingJobName=mxnet_training_job)
+            model_data = desc['ModelArtifacts']['S3ModelArtifacts']
+            script_path = os.path.join(DATA_DIR, 'mxnet_mnist', 'mnist.py')
+            model = MXNetModel(model_data, 'SageMakerRole', entry_point=script_path,
+                               py_version=PYTHON_VERSION, sagemaker_session=sagemaker_session)
+            model.deploy(1, 'ml.t2.medium', endpoint_name=endpoint_name)
+            sagemaker_session.describe_endpoint(EndpointName=endpoint_name)
+            model.deploy(1, 'ml.m4.xlarge', update_endpoint=True, endpoint_name='non-existing-endpoint')
+        except ValueError:
```

Comment:
I think using the pytest.raises might look cleaner here. https://github.com/aws/sagemaker-python-sdk/blob/900717f348087121afc731a3d1e0825451137730/tests/unit/test_session.py#L891

Let's assert that the exception has the right error message.